Scripts/XsvReader.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Monry.XsvUtility
+{
+    [Serializable]
+    public class XsvReader
+    {
+        [SerializeField, Tooltip("Choose delimiter, CSV->Comma, TSV->Tab")]
+        public XsvParser.Delimiter m_Delimiter = XsvParser.Delimiter.Comma;
```

Comment:
```suggestion
        private XsvParser.Delimiter delimiter = XsvParser.Delimiter.Comma;
        public XsvParser.Delimiter Delimiter => delimiter;
```

フィールド名の変更に弱くなってしまうのと、意図しない変更を検知しづらいため、 public フィールドを直接利用するのは避けた方が良いです。
（ここで言う「意図しない変更」は、他のクラスからの変更を指します。今回の実装で言うと、Inspector からの設定を前提としているはずですので、上記の Suggest が望ましいと考えます。）

Reply:
また、Microsoft が提案する C# の名前付けガイドラインに従い、フィールド名の接頭辞 `m_` を外したいです。
（厳密に言うと private および internal なフィールドについては言及していないのですが、ルールを統一しておきたいです。）
https://docs.microsoft.com/ja-jp/dotnet/standard/design-guidelines/names-of-type-members#names-of-fields

Reply:
自動で命名規則をハンガリアン記法にするような仕組みがあって、むしろ推奨してるのかと思ってそういう記法を使っていました。
今調べて現在は推奨していないとわかったので外すようにします。
Suggestで設定する方法も今回初めて知りました（GetSetだけじゃなかったと驚いてます）
今後はその形で実装してみます、親切にありがとうございます…！

Reply:
相談ですが、メンバ変数を変更して実装する際に読み取り専用で変更できません。こちらとしてはInspectorからだけを想定してるわけではありません。
しかしこちらの想定ではpublicフィールドを使用するかゲッターを挟むかしかないので、Getterを含むプロパティとして指定することにします。
あらかじめご了承ください。